data/provider/src/main/java/tech/pegasys/teku/api/NetworkDataProvider.java

```diff
@@ -97,4 +101,11 @@ public Metadata getMetadata() {
 
     return new tech.pegasys.teku.api.response.v1.node.Peer(peerId, null, address, state, direction);
   }
+
+  private <R> tech.pegasys.teku.api.response.v1.node.PeerScore toPeerScore(final Eth2Peer eth2Peer) {
+    final String peerId = eth2Peer.getId().toBase58();
+    final String gossipScore = String.valueOf(eth2Peer.getGossipScore());
```

Comment:
Directly translating to a string can create some fairly unreadable values.  We might want to format this value by rounding to 4 decimal points or something ...  You can test by running the teku with `--Xp2p-gossip-scoring-enabled=true` to get some non-zero scoring values.  It's probably fine for now - just putting this up for discussion.

Reply:
Looked into this some more, I think what we actually want to do is to keep `gossipScore` stored as a `Double`.  And then add some custom serializers to handle doubles (where we can apply nicer formatting).  You can then register those (de)serializers in `JsonProvider`:  [example here](https://github.com/ConsenSys/teku/blob/e97a6679902ef7ae353bccad5b9508f0c15ca4d2/data/serializer/src/main/java/tech/pegasys/teku/provider/JsonProvider.java#L49-L50).

Reply:
I kept `gossipScore` stored as `Double`, and I've added a `DoubleSerializer`.